test.js

```diff
@@ -26,3 +26,23 @@ test('promise-map should map a fn', function(t) {
     t.fail('catch')
   }
 })
+
+test('promise-map should map a fn that returns promises', function(t) {
+  t.plan(1)
+  const res = Promise.resolve([2, 3, 4])
+    .then(map(mapFn))
+    .then(checkFn)
+    .catch(handleErr)
+
+  function mapFn(val) {
+    return Promise.resolve(val + 1);
+  }
+
+  function checkFn(val) {
+    t.deepEqual([3, 4, 5], val)
+  }
+
+  function handleErr() {
+    t.fail('catch')
+  }
```

Comment:
there's no need for this function; it lowers code cov and if it fails we'll know. Thanks!